### ISSUE TITLE
Debugger stops at HALT

### ISSUE THREAD

**Issue by stuurmn** [Dotneteer/spectnetide#60]
#46 
Seems to have re appeared? (48k)
In the emulator the program is running fine.
In the disassembly window and the code window the cursor is stuck at the halt statement.
Also in the toolwindow the PC is stuck at the same address as the halt statement

**Comment by Dotneteer** [comment]
Albert, I did not manage to reproduce this behavior. In my code samples `HALT` behaves as expected.
However, there's a way `HALT` may produce the behavior you described. When the interrupt is disabled (with `DI`), the `HALT` stays forever, as this is its nature. In the following code snippet, when you use `DI`, you cannot get away from `HALT`. If you comment out `DI`, stepping over `HALT` works.

```
start: .org #8000
    ld b,100
    di
wait:
    halt
    djnz wait
    ld a,2
    out (#fe),a
    jp #12a2
```
If you can send me a sample code where you can reproduce this issue, I will certainly examine that.

**Comment by stuurmn** [comment]
Istvan,
Things are  ***very*** **slowly** starting to make sence.
The "Step Over (F10)" is the answer. I had no experience what so ever with interupts.
Now I can jump over the halt and it does make sence.

But this is not the hole story.
Something goes wrong when I play around for longer. (The problem: I am not able to reproduce it consistenly) But it does happen more often.
The result is that I have to fully close the Visual Studio Experimental  and restart it to get the emulator / debugger working again properly (Just shutting down the tool windows is not enough)

Some of the symptoms I get while debugging:


 I'm getting errors like.

![2017-12-23 16_00_44-demo_libzx - microsoft visual studio - experimental instance](https://user-images.githubusercontent.com/33286419/34320784-757c4d20-e801-11e7-939a-ba071b57da01.jpg)

or (Seen only once)

[ActivityLog.zip](https://github.com/Dotneteer/spectnetide/files/1584281/ActivityLog.zip)
![2017-12-23 16_03_09-demo_libzx - microsoft visual studio - experimental instance](https://user-images.githubusercontent.com/33286419/34321155-c48306aa-e808-11e7-97f4-7100e04b89d5.jpg)


or

The wrong Debug buttons where active

I have had once that in the included project that I started example x and it was trying to run example y


I have got 1 example which I have been able to reproduce
In the included project there is an example5. "Color Cycling" When I run that and go into debugging. It stops at the halt command.

Now I know with Step Over it goes further. and with more Step Overs (F10) it does step through the code.
 When I use a combination of F5 F10 F11, then after a while  the result is that F11 does the same thing as F5 (Start debugging). So the program starts running again while I expected it would only do a step forward.

I know they are very vague errors....
I'm just hoping that you recognise the behaviour or that there is a that buffer that you can increase a buffer size otherwise this is a fairly hopeless error to search for

(But I'll keep trying.....)


[Demo_libzx.zip](https://github.com/Dotneteer/spectnetide/files/1584264/Demo_libzx.zip)

**Comment by Dotneteer** [comment]
Thanks for the info, I try to reproduce the issue.

**Comment by Dotneteer** [comment]
Albert, I managed to reproduce the issue that the program does not stop, just carries on running. Now I'm working on finding the root cause and fixing it.

**Comment by Dotneteer** [comment]
The issue was easy to fix. Please, open a new issue next time -- should the issues in the pictures come again -- with the explicit phenomenon. Merry Christmas!
